python/pyarrow/tests/parquet/test_parquet_encryption.py

```diff
@@ -0,0 +1,248 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pyarrow as pa
+import pytest
+
+try:
+    import pyarrow.parquet as pq
+except ImportError:
+    pq = None
+
+import base64
+from Cryptodome.Cipher import AES
+from collections import OrderedDict
+from datetime import timedelta
+
+PARQUET_NAME = 'encrypted_table.in_mem.parquet'
+FOOTER_KEY = "0123456789ABCDEF"
+FOOTER_KEY_NAME = "footer_key"
+COL_KEY = "1234123412341234"
+COL_KEY_NAME = "col_key"
+DATA_TABLE = pa.Table.from_pydict(
+    OrderedDict([
+        ('a', pa.array([1, 2, 3])),
+        ('b', pa.array(['a', 'b', 'c'])),
+        ('c', pa.array(['x', 'y', 'z']))
+    ])
+)
+
+
+class InMemoryKmsClient(pq.KmsClient):
```

Comment:
Can you also add a test with a KmsClient that would raise an error from `wrap_key` or `unwrap_key`?

Reply:
Yes, thank you @pitrou , that's what test_encrypted_parquet_write_kms_error() does - since it initializes the InMemoryKmsClient with an empty master keys map, a KeyError will be raised on` wrap_key()` (and `unwrap_key()`) calls.

Reply:
Hmm, perhaps, but it would also be nice to test that any custom error is also propagated (including the Python error message).